tests/cli/device/test_deploy_and_device.py

```diff
@@ -0,0 +1,174 @@
+                assert device.id in result.output
+
+        # Verify list-applications command list default android apps and the app we've just deployed
+        for device in self.ANDROID_DEVICES:
+            result = Tns.exec_command('device list-applications --device {0}'.format(device.id))
+            assert 'com.android' in result.output
+            assert app_id in result.output
+
+        # Verify `device run <bundle-id>` will start the app
+        device = self.ANDROID_DEVICES[0]
+        Adb.stop_application(device_id=device.id, app_id=app_id)
+        assert not device.is_text_visible(text=Changes.JSHelloWord.JS.old_value), 'Failed to stop the app.'
+        result = Tns.exec_command(command='device run {0}'.format(app_id), device=device.id, wait=True,
+                                  just_launch=True)
+        if Settings.HOST_OS == OSType.WINDOWS:
+            assert 'iTunes is not installed. Install it on your system and run this command again.' in result.output
+        device.wait_for_text(text=Changes.JSHelloWord.JS.old_value)
+
+        # Get Android device logs
+        result = Tns.exec_command(command='device log', device=device.id, wait=False)
+        TnsLogs.wait_for_log(log_file=result.log_file, string_list=['beginning of'], timeout=120)
+        assert 'I' or 'D' or 'W' in File.read(result.log_file), 'Log does not contain INFO, DEBUG or WARN messages.'
+
+        # Get iOS device logs
+        if Settings.HOST_OS == OSType.OSX:
+            device = self.IOS_DEVICES[0]
+            result = Tns.exec_command(command='device log', device=device.id, wait=False)
+            TnsLogs.wait_for_log(log_file=result.log_file, string_list=['>:'], timeout=120)
+            assert "<Notice>:" or "<Error>:" in File.read(result.log_file), 'tns device log fails to get ios logs.'
+
+    @unittest.skipIf(Settings.HOST_OS == OSType.WINDOWS, 'Skip on Windows machines.')
+    def test_310_device_log_android_two_devices(self):
+        if len(self.ANDROID_DEVICES) + len(self.IOS_DEVICES) > 2:
+            result = Tns.exec_command('device log')
+            assert 'More than one device found. Specify device explicitly.' in result.output
+
+    def test_400_device_and_deploy_invalid_platform(self):
+        invalid_platform = 'fakePlatform'
+        result = Tns.exec_command(command='device {0}'.format(invalid_platform))
+        assert "'{0}' is not a valid device platform.".format(invalid_platform) in result.output
+
+    def test_401_device_commands_with_invalid_device_id(self):
+        invalid_device_id = 'fakeDevice'
+        for command in ['log', 'run android', 'list-applications']:
+            output = Tns.exec_command('device {0} --device {1}'.format(command, invalid_device_id)).output
+            assert "Could not find device by specified identifier '{0}'.".format(invalid_device_id) in output
+            assert "To list currently connected devices and verify that the specified identifier exists" in output
```

Comment:
This is already covered with unit tests: https://github.com/NativeScript/nativescript-cli/blob/1760db353cb34579083061acb8393da6b0b68b4e/lib/common/test/unit-tests/mobile/devices-service.ts#L531-L546

Reply:
Great will remove e2e test.